### PR TITLE
Create #1351_Count_Negative_Numbers_in_a_Sorted_Matrix.py

### LINKED ISSUE #3: #1351_Count_Negative_Numbers_in_a_Sorted_Matrix
# Summary
add 1351. Count Negative Numbers in a Sorted Matrix

# Goal
add 1351. Count Negative Numbers in a Sorted Matrix

# Todo
- [ ] add 1351. Count Negative Numbers in a Sorted Matrix
- [ ] time  complexity: O(n^2)
- [ ] space complexity: 

# Deadline
/

# Parent issue
# References
None

# Notes
None

### PR DESCRIPTION
**Pull request by ashizawaN**
Closes #3

### CODE REVIEW

File: #1351_Count_Negative_Numbers_in_a_Sorted_Matrix.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Solution:
+    def countNegatives(self, grid: List[List[int]]) -> int:
+        count=0
+        
+        for i in range(len(grid)):
+            for j in range(len(grid[i])):
+                if grid[i][j]<0:
```
Comment (nayuta-ai) [inline #1351_Count_Negative_Numbers_in_a_Sorted_Matrix.py:7]: if grid[i][j] < 0:

File: #1351_Count_Negative_Numbers_in_a_Sorted_Matrix.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Solution:
```
Comment (nayuta-ai) [inline #1351_Count_Negative_Numbers_in_a_Sorted_Matrix.py:1]: Rewrite it by using binary search.

**Review by nayuta-ai — CHANGES_REQUESTED** [review]

File: #1351_Count_Negative_Numbers_in_a_Sorted_Matrix.py (hunk 2 above)
Comment (ashizawaN) [reply #1351_Count_Negative_Numbers_in_a_Sorted_Matrix.py:3]: I rewote code with binary search. The test was passed, but when I submitted it, I got the following error. Could you give me some hints to revise it?

## Error

```
Runtime Error Message:
IndexError: list index out of range
    if num[m] < 0:
Line 11 in BinarySearch (Solution.py)
    count += BinarySearch(x)
Line 21 in countNegatives (Solution.py)
    ret = Solution().countNegatives(param_1)
Line 41 in _driver (Solution.py)
    _driver()
Line 52 in <module> (Solution.py)
```

## Code
```
from typing import List

class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:    
        def BinarySearch(num):
            left = 0
            right = len(num)
        
            while left <= right:
                m = (left+right) // 2
                if num[m] < 0:
                    right = m - 1
                elif num[m] > 0:
                    left = m + 1
                else:
                    return m
            return left
        
        count = 0
        for x in grid:
            count += BinarySearch(x)
        return count
```

File: #1351_Count_Negative_Numbers_in_a_Sorted_Matrix.py [hunk 3]
```diff
@@ -1,22 +1,23 @@
 from typing import List
 
+
 class Solution:
     def countNegatives(self, grid: List[List[int]]) -> int:    
         def BinarySearch(num):
```
Comment (nayuta-ai) [inline #1351_Count_Negative_Numbers_in_a_Sorted_Matrix.py:6]: Use snake-case for naming methods.

**Review by nayuta-ai — CHANGES_REQUESTED** [review]

**Review by nayuta-ai — APPROVED** [review]

**Pull request merged**